applications/HDF5Application/custom_utilities/data_type_utilities.h

```diff
@@ -0,0 +1,50 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:        BSD License
+//                  license: HDF5Application/license.txt
+//
+//  Main author:    Suneth Warnakulasuriya
+//
+
+// System includes
+#include "hdf5.h"
+#include <type_traits>
+
+// Project includes
+#include "utilities/data_type_traits.h"
+
+// Application includes
+
+
+namespace Kratos
+{
+namespace HDF5
+{
+namespace Internals
+{
+
+// H5 data types
+template <class TDataType> hid_t inline GetPrimitiveH5Type()
```

Comment:
it's a header-only template so you don't need the `inline`.
```suggestion
template <class TDataType> hid_t GetPrimitiveH5Type()
```

Reply:
done :)